### DISCUSSION TITLE
Change file mode for web storage

### DISCUSSION THREAD

**Discussion by Programie** [SpectrumIM/spectrum2#432] [Ideas]
I've just configured Web Storage for the Facebook transport of my Spectrum 2 installation. And it just works... with a tiny but annoying issue: The files are saved with mode 0660 so only the owner and group can access the files.

That's an issue in my case as I'm using a NFS mount for the web storage directory and mapping all users to nobody/nogroup (all_squash NFS export option). My web server is running on a different host under a different user than nobody or nogroup (www-data) and therefore can't access the files stored by Spectrum.

Is there any way to configure the file mode which is used for files saved by Spectrum?

**Comment by vitalyster** [comment]
1. Why not add www-data to nogroup?
2. https://github.com/SpectrumIM/spectrum2/blob/master/spectrum/src/main.cpp#L174 umask 0007 make 0660 permissions. I'm not sure why 0007 is selected, but if you know how to read default umask for the current user we can change it

**Comment by Programie** [comment]
I already thought about adding www-data to nogroup while creating this issue. Maybe that's the easiest way.

As far as I understand, `umask()` returns the old value. So it should be possible to change umask to a custom value from config and revert it after saving the file.

But why is Spectrum even changing the umask? Shouldn't the umask kept on the default value to allow the user to change it if required?
